embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/PendingApiCalls.kt

```diff
@@ -17,11 +20,11 @@ internal class PendingApiCalls {
      */
     fun add(pendingApiCall: PendingApiCall) {
         val endpoint = pendingApiCall.apiRequest.url.endpoint()
-        val pendingApiCallsForEndpoint = pendingApiCallsMap.getOrPut(endpoint) { PendingApiCallsQueue() }
+        val pendingApiCallsForEndpoint = pendingApiCallsMap.getOrPut(endpoint, ::mutableListOf)
```

Comment:
Is there a concern that `ArrayList` isn't thread safe? Or are access into a particular queue run serially so there's never multiple threads access it?

Reply:
I don't think that the fact of `ArrayList` not being thread-safe could be a problem here, as we are synchronizing manually on each access to the lists. 
The lock is held on each list so different lists can be accessed at the same time.